crates/bevy_pbr/src/deferred/mod.rs

```diff
@@ -0,0 +1,451 @@
+            .add_systems(
+                Render,
+                (prepare_deferred_lighting_pipelines.in_set(RenderSet::Prepare),),
+            )
+            .add_render_graph_node::<ViewNodeRunner<DeferredLightingNode>>(
+                core_3d::graph::NAME,
+                DEFERRED_LIGHTING_PASS,
+            )
+            .add_render_graph_edges(
+                core_3d::graph::NAME,
+                &[
+                    core_3d::graph::node::START_MAIN_PASS,
+                    DEFERRED_LIGHTING_PASS,
+                    core_3d::graph::node::MAIN_OPAQUE_PASS,
+                ],
+            );
+    }
+
+    fn finish(&self, app: &mut App) {
+        let Ok(render_app) = app.get_sub_app_mut(RenderApp) else {
+            return;
+        };
+
+        render_app.init_resource::<DeferredLightingLayout>();
+    }
+}
+
+pub const DEFERRED_LIGHTING_PASS: &str = "deferred_lighting";
+
+#[derive(Default)]
+pub struct DeferredLightingNode;
+
+impl ViewNode for DeferredLightingNode {
+    type ViewQuery = (
+        &'static ViewUniformOffset,
+        &'static ViewLightsUniformOffset,
+        &'static ViewFogUniformOffset,
+        &'static MeshViewBindGroup,
+        &'static ViewTarget,
+        &'static DeferredLightingIdDepthTexture,
+        &'static Camera3d,
+        &'static DeferredLightingPipeline,
+    );
+
+    fn run(
+        &self,
+        _graph_context: &mut RenderGraphContext,
+        render_context: &mut RenderContext,
+        (
+            view_uniform_offset,
+            view_lights_offset,
+            view_fog_offset,
+            mesh_view_bind_group,
+            target,
+            deferred_lighting_id_depth_texture,
+            camera_3d,
+            deferred_lighting_pipeline,
+        ): QueryItem<Self::ViewQuery>,
+        world: &World,
+    ) -> Result<(), NodeRunError> {
+        let pipeline_cache = world.resource::<PipelineCache>();
+        let deferred_lighting_layout = world.resource::<DeferredLightingLayout>();
+
+        let Some(pipeline) =
+            pipeline_cache.get_render_pipeline(deferred_lighting_pipeline.pipeline_id)
+        else {
+            return Ok(());
+        };
+
+        let deferred_lighting_depth_id =
+            world.resource::<ComponentUniforms<PbrDeferredLightingDepthId>>();
+        let Some(deferred_lighting_depth_id_binding) =
+            deferred_lighting_depth_id.uniforms().binding()
+        else {
+            return Ok(());
+        };
+
+        let bind_group_1 = render_context
+            .render_device()
+            .create_bind_group(&BindGroupDescriptor {
+                label: Some("deferred_lighting_layout_group_1"),
+                layout: &deferred_lighting_layout.bind_group_layout_1,
+                entries: &[BindGroupEntry {
+                    binding: 0,
+                    resource: deferred_lighting_depth_id_binding.clone(),
+                }],
+            });
```

Comment:
Does this need to be created per-frame? Seems like if the deferred lighting depth id binding doesn't change then the bind group doesn't need to be recreated.